optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -108,8 +109,9 @@ FeatureBuildItem feature() {
     }
 
     @BuildStep
-    void registerScoreDirectorFactorySpi(BuildProducer<ServiceProviderBuildItem> services) {
-        registerSpi(ScoreDirectorFactoryService.class, services);
+    void registerSpi(BuildProducer<ServiceProviderBuildItem> services) {
+        Stream.of(ScoreDirectorFactoryService.class, JoinerService.class)
+                .forEach(service -> registerSpi(service, services));
```

Comment:
(nitpick) Stream seems like an overkill here. You could simply
```java
registerSpi(ScoreDirectorFactoryService.class, services);
registerSpi(JoinerService.class, services);
```

Reply:
I see your point. In the interest of fixing the build quickly, I will keep it as is, but I will remember it for when I surely touch this code in the future.

Reply:
No big deal :slightly_smiling_face: